libnavigation-core/src/test/java/com/mapbox/navigation/core/routerefresh/RouteRefreshControllerTest.kt

```diff
@@ -570,9 +621,394 @@ class RouteRefreshControllerTest {
+        }
+
+    @Test
+    fun `if route refresh works only for one route, the controller updates only one route`() =
+        runBlockingTest {
+            val currentTime = utcToLocalTime(
+                year = 2022,
+                month = Month.MAY,
+                date = 22,
+                hourOfDay = 12,
+                minute = 30,
+                second = 0
+            )
+            val routeRefreshStub = RouteRefreshStub()
+            val initialRoutes = createNavigationRoutes(
+                createDirectionsResponse(
+                    routes = listOf(
+                        createTestTwoLegRoute(),
+                        createTestTwoLegRoute(
+                            secondLegIncidents = listOf(
+                                createIncident(
+                                    id = "1",
+                                    endTime = "2022-05-21T14:00:00Z", // expired
+                                ),
+                                createIncident(
+                                    id = "2",
+                                    endTime = "2022-05-22T14:00:00Z",
+                                ),
+                            )
+                        )
+                    )
+                )
+            )
+            val refreshedRoutes = createNavigationRoutes(
+                createDirectionsResponse(
+                    routes = listOf(
+                        createTestTwoLegRoute(
+                            firstLegAnnotations = createRouteLegAnnotation(
+                                congestion = listOf("severe", "severe"),
+                                congestionNumeric = listOf(90, 99),
+                            )
+                        ),
+                        createTestTwoLegRoute()
+                    )
+                )
+            )
+            routeRefreshStub.setRefreshedRoute(refreshedRoutes[0])
+            routeRefreshStub.doNotRespondForRouteRefresh(refreshedRoutes[1].id)
+            val refreshOptions = RouteRefreshOptions.Builder().build()
+            val routeRefreshController = createRouteRefreshController(
+                routeRefresh = routeRefreshStub,
+                routeRefreshOptions = refreshOptions,
+                localDateProvider = { currentTime },
+                currentLegIndexProvider = { 1 }
+            )
+
+            val refreshedRoutesDeferred = async {
+                routeRefreshController.refresh(initialRoutes)
+            }
+            advanceTimeBy(
+                expectedTimeToInvalidateCongestionsInCaseOfTimeout(refreshOptions.intervalMillis)
+            )
+
+            val result = refreshedRoutesDeferred.getCompletedTest()
+
+            assertEquals(
+                refreshedRoutes[0],
+                result[0]
+            )
+            assertEquals(
+                initialRoutes[1], // no cleanup happens
+                result[1]
+            )
+        }
```

Comment:
I'm not sure about this case. If one route always fails to refresh, should route refresh controller cleanup congestion data/incidents from it like the controller does if all routes refresh fails?
@LukasPaczos , @RingerJK , @mapbox/navigation-android , WDYT?

Reply:
At this point my opinion is that the UX should be consistent at all times since that boosts the confidence in the data, even if the correctness is not always there. If a route fails to refresh, whether we clean it up or not doesn't change the fact that the result is incorrect. That's why I'd say that we should only clean up all the routes together not independently.

Reply:
Looks like we disagree 🙃 I would still advocate for a consistent experience though - either we clean up all congestion, or we clean up none. The primary driver for this feature was consistency with the map, and the map won't clean up congestion if only one alternative request fails, it would only clean up congestion when there's no connectivity which aligns with all routes failing to refresh, and only then we clean up the stale congestion.

Reply:
primary route refresh failed 3 times, all alternatives routes refreshed successfully - primary stays the same, alternatives updated

otherwise looks good, that'd be my proposal.